src/webapi/date.rs

```diff
@@ -1,25 +1,1166 @@
-use std::marker::PhantomData;
-use webcore::try_from::TryInto;
+use std::collections::HashMap;
+use webcore::try_from::{TryFrom, TryInto};
+use webcore::value::{Value, Reference};
 
-/// [(JavaScript docs)](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date)
-#[derive(Debug)]
-pub struct Date {
-    dummy: PhantomData< () >
-}
-
-impl Date {
-    /// The Date.now() method returns the number of milliseconds elapsed since 1 January 1970 00:00:00 UTC.
+/// A date object
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Date)
+pub trait IDate: AsRef< Reference > + TryFrom< Value >  {
```

Comment:
Does anything actually inherit from Date? (I don't think so, but I might be missing something.) If that's the case then I don't see a point in making an interface out of it; just put all of the methods in an `impl Date`.

Reply:
Correct me if I'm wrong, but does Date need to be a trait in order to coerce a ```js!``` statement into a date "object"? For example, if Date was an ```impl```, would we still be able to do this?

```rust
let d: Date = js!(
    return new Date();
)
```

Reply:
It doesn't need to be a trait. Please take a look at, for example, `array_buffer.rs` for a minimal example on how to do it.